File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.py
# N 마리 중 N/2 마리 가져갈 수 있음
# [3,1,2,3] 주어지면
# 3,1 / 3,2 / 3,3 / 1,2 / 1,3 / 2,3

def solution(nums):
    go=len(nums)//2
    unique=len(set(nums))
    answer = min(go,unique)
    return answer

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.java
import java.util.*;

class Solution {
    public int solution(int[] nums) {
        Set<Integer> types = new HashSet<>();
        for (int num : nums) {
            types.add(num);
        }
        
        int canPick = nums.length / 2;
        int uniqueCount = types.size();
        
        return Math.min(canPick, uniqueCount);
    }
}